extensions/commands/install-universal/cmd_install_universal.py

```diff
@@ -0,0 +1,248 @@
+    gapi = conan_api.graph
+    if path:
+        deps_graph = gapi.load_graph_consumer(path, args.name, args.version, args.user, args.channel,
+                                              profile_host, profile_build, lockfile, remotes,
+                                              args.update, is_build_require=args.build_require)
+    else:
+        deps_graph = gapi.load_graph_requires(args.requires, args.tool_requires, profile_host,
+                                              profile_build, lockfile, remotes, args.update)
+
+    # Handle universal packages
+    if profile_host.settings['os'] in ['Macos', 'iOS', 'watchOS', 'tvOS', 'visionOS'] and '|' in profile_host.settings['arch']:
+        for node in deps_graph.ordered_iterate():
+            make_universal_conanfile(node.conanfile, args)
+            print(node.conanfile.name, node.conanfile.generate, node.conanfile.build)
+
+    print_graph_basic(deps_graph)
+    deps_graph.report_graph_error()
+    gapi.analyze_binaries(deps_graph, args.build, remotes, update=args.update, lockfile=lockfile)
+    print_graph_packages(deps_graph)
+
+    # Installation of binaries and consumer generators
+    conan_api.install.install_binaries(deps_graph=deps_graph, remotes=remotes)
+    ConanOutput().title("Finalizing install (deploy, generators)")
+    conan_api.install.install_consumer(deps_graph, args.generator, source_folder, output_folder,
+                                       deploy=args.deployer, deploy_package=args.deployer_package,
+                                       deploy_folder=args.deployer_folder,
+                                       envs_generation=args.envs_generation)
+    ConanOutput().success("Install finished successfully")
+
+    # Update lockfile if necessary
+    lockfile = conan_api.lockfile.update_lockfile(lockfile, deps_graph, args.lockfile_packages,
+                                                  clean=args.lockfile_clean)
+    conan_api.lockfile.save_lockfile(lockfile, args.lockfile_out, cwd)
+    return {"graph": deps_graph,
+            "conan_api": conan_api}
+
+
+def make_universal_conanfile(conanfile, args):
+    def _generate(conanfile):
+        print('generate', conanfile.name)
+        pass
+    def _build(conanfile):
+        print('build', conanfile.name)
+        archs = str(conanfile.settings.arch).split('|')
+        ref = conanfile.ref
+        for arch in archs:
+            more_args = ''
+            if args.profile_build:
+                more_args += f' -pr:b {args.profile_build[0]}'
+            if args.profile_host:
+                more_args += f' -pr:h {args.profile_host[0]}'
+            for arch in archs:
+                ConanOutput().success(f"Building universal {ref} {arch}")
+                arch_folder = os.path.join(conanfile.build_folder, arch)
+                run(f'conan install --requires={ref}{more_args} -d direct_deploy --deployer-folder "{arch_folder}" -s arch={arch} -b missing', shell=True)
```

Comment:
This is making Conan re-entrant, and it is not supported, see https://docs.conan.io/2/knowledge/guidelines.html -> Forbidden practices.

> Conan is not re-entrant: Calling the Conan process from Conan itself cannot be done. That includes calling Conan from recipe code, hooks, plugins, and basically every code that already executes when Conan is called. Doing it will result in undefined behavior.

Reply:
Well, this isn't quite resolved. It can fail due to skipped binaries but I'll give it some thought.